plugins/nf-amazon/src/test/nextflow/cloud/aws/batch/AwsBatchTaskHandlerTest.groovy

```diff
@@ -1100,4 +1102,18 @@ class AwsBatchTaskHandlerTest extends Specification {
         'job1'       | 'job1'
         'job1:task2' | 'job1'
     }
+
+    def 'should include tower workflow id in job name'() {
+        given:
+        def handlerWithEnv = new AwsBatchTaskHandler(environment: [TOWER_WORKFLOW_ID: '1234'])
+        def handlerWithoutEnv = new AwsBatchTaskHandler()
+        def name = 'job_1'
+        when:
+        def convertedName = handlerWithEnv.includeTowerPrefix(name)
+        def nonConvertedName = handlerWithoutEnv.includeTowerPrefix(name)
+        then:
+        convertedName == "tw-1234-job_1"
+        nonConvertedName == "job_1"
```

Comment:
This is OK, but it can be written using a more idiotic approach with `where:` something like: 

```
where: 
ENV                          | EXPECTED
[TOWER_WORKFLOW_ID: '1234']  |  "tw-1234-job_1"
[:]                          | "job_1"
```

there are plenty of examples look around

Reply:
Changed